homeassistant/components/homekit/__init__.py

```diff
@@ -437,7 +444,7 @@ async def async_handle_homekit_unpair(service: ServiceCall) -> None:
             matching_instances = [
                 homekit
                 for homekit in _async_all_homekit_instances(hass)
-                if homekit.driver
+                if hasattr(homekit, "driver")
```

Comment:
The old check would also guard for `homekit.driver` being set to `None`. Is that still needed?

Reply:
I refactored it to avoid the `hasattr`.  I had to add some asserts instead since `self.driver` has to be set in the executor since the underlying constructor does I/O in the library.

b04dd6f8302dbf38f337a7d6066c224f0168c71d